src/utils.js

```diff
@@ -236,6 +236,84 @@ function camelcase (dashCaseText) {
     })
 }
 
+// We use this method to detect M1 macs and set appropriate API values to prevent sites from detecting fingerprinting protections
+function isAppleSilicon () {
+    const canvas = document.createElement('canvas')
+    const gl = canvas.getContext('webgl')
+
+    // Best guess if the device is an Apple Silicon
+    // https://stackoverflow.com/a/65412357
+    return gl.getSupportedExtensions().indexOf('WEBGL_compressed_texture_etc') !== -1
+}
+
+/**
+ * Take configSeting which should be an array of possible values.
+ * If a value contains a criteria that is a match for this environment then return that value.
+ * Otherwise return the first value that doesn't have a criteria.
+ *
+ * @param {array} configSetting - Config setting which should contain a list of possible values
+ * @returns - The value from the list that best matches the criteria in the config
+ */
+function processAttrByCriteria (configSetting) {
+    if (!configSetting.some(item => item.criteria !== undefined)) {
+        return configSetting
+    }
+
+    let bestOption = configSetting[0]
+    for (const item of configSetting) {
+        if (item.criteria) {
+            if (item.criteria.arch === 'AppleSilicon' && isAppleSilicon()) {
+                bestOption = item
+                break
+            }
+        } else {
+            bestOption = item
+        }
+    }
+
+    return bestOption
+}
+
+/**
+ * Get the value of a config setting.
+ * If the value is not set, return the default value.
+ * If the value is not an object, return the value.
+ * If the value is an object, check its type property.
+ *
+ * @param {string} featureName
+ * @param {object} args
+ * @param {string} prop
+ * @param {any} defaultValue - The default value to use if the config setting is not set
+ * @returns The value of the config setting or the default value
+ */
+export function getFeatureAttr (featureName, args, prop, defaultValue) {
+    let configSetting = getFeatureSetting(featureName, args, prop)
+
+    if (configSetting === undefined) {
+        return defaultValue
+    }
+
+    const configSettingType = typeof configSetting
+    switch (configSettingType) {
+    case 'object':
+        if (Array.isArray(configSetting)) {
+            configSetting = processAttrByCriteria(configSetting)
+        }
+
+        if (!configSetting.type) {
+            return configSetting
+        }
+
+        if (configSetting.type === 'undefined') {
+            return undefined
+        }
+
+        return configSetting.value
+    default:
+        return configSetting
```

Comment:
Should this be _defaultValue_?

Reply:
Same thinking as above. Will change to `defaultValue`